be/src/exec/hash_join_node.cpp

```diff
@@ -484,24 +470,36 @@ pipeline::OpFactories HashJoinNode::decompose_to_pipeline(pipeline::PipelineBuil
     // In default query engine, we only build one hash table for join right child.
     // But for pipeline query engine, we will build `num_right_partitions` hash tables, so we need to enlarge the limit
     std::unique_ptr<PartialRuntimeFilterMerger> partial_rf_merger = std::make_unique<PartialRuntimeFilterMerger>(
-            pool, _runtime_join_filter_pushdown_limit * num_right_partitions, num_right_partitions);
```

Comment:
no num_right_partitions passed into PartialRuntimeFilterMerger, how does it known from which it counts down?

Reply:
Add a new method `incr_builder` to `PartialRuntimeFilterMerger`, which is invoked by `HashJoinBuildOperator::prepare`. Could see [diffs in runtime_filter_types.h](https://github.com/StarRocks/starrocks/pull/18152/files#diff-203ba9013a08420ee7f2601cc7633f5dbb4a65da2e3fd2c80ccdd2a57cb45945) for detail.